Решение финального задания 9 спринта

RecoilSPB opened the pull request.

RecoilSPB committed:
Решение финального задания 9 спринта

z8q commented on src/ru/practicum/http/TaskHttpServer.java:
```diff
@@ -0,0 +1,92 @@
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class TaskHttpServer {
```
Достаточно же 1-го сервера, зачем второй?

z8q commented on src/ru/practicum/http/HttpTaskServer.java:
```diff
@@ -0,0 +1,353 @@
+    public HttpTaskServer(TaskManager taskManager) throws IOException {
+        this.taskManager = taskManager;
+        httpServer = HttpServer.create(new InetSocketAddress("localhost", PORT), 0);
+        httpServer.createContext("/tasks", new TasksHandler());
```
стоит разбить на несколько хэндлеров
```JAVA
        httpServer.createContext("/tasks", new TasksHandler(manager));
        httpServer.createContext("/subtasks", new SubtasksHandler(manager));
        httpServer.createContext("/epics", new EpicsHandler(manager));
        httpServer.createContext("/history", new HistoryHandler(manager));
        httpServer.createContext("/prioritized", new PrioritizedHandler(manager));
```

Сами хэндлеры лучше вынести в отдельные классы

z8q commented on src/ru/practicum/service/HttpTaskManager.java:
```diff
@@ -0,0 +1,93 @@
+
+import java.util.stream.Collectors;
+
+public class HttpTaskManager extends FileBackedTaskManager {
```
Не уверен, будут ли нужны HttpTaskManager и HttpTaskClient, можно посоветоваться с наставником.
Из тз:
```
В тестах для HttpTaskServer используйте InMemoryTaskManager в качестве зависимости для конструктора HttpTaskServer, чтобы имитировать реальное хранение и обработку данных.
```
Возможно мысль создать HttpTaskManager пришла исходя из названия тестового класса в примерах тз – HttpTaskManagerTasksTest



Если предположим удалили HttpTaskManager и клиент, то в тесте будем создавать сервер с InMemoryTaskManager, стартовать его,
И далее вызовы мы можем строить таким образом
```JAVA
        HttpClient client = HttpClient.newHttpClient();
        URI url = URI.create("http://localhost:8080/tasks");
        HttpRequest request = HttpRequest.newBuilder().uri(url).GET().build();
        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
```

z8q commented on test/ru/practicum/service/HttpTaskManagerTest.java:
```diff
@@ -0,0 +1,72 @@
+    @BeforeEach
+    void setUp() {
+        try {
+            server = new TaskHttpServer();
```
В конструктор сервера надо передавать экземпляр менеджера

RecoilSPB committed:
[-] Убрал один сервер (TaskHttpServer)
[-] Убрал клиент (HttpTaskClient)
[-] Убрал менеджера (HttpTaskManager)
[+] Добавил справочник Endpoint
[!] Исправил тесты в соответсвии с изменениями

RecoilSPB committed:
[!] Разнес Handlers на разные классы.
